k2/csrc/aux_labels.cc

```diff
@@ -0,0 +1,66 @@
+// k2/csrc/aux_labels.cc
+
+// Copyright (c)  2020  Xiaomi Corporation (author: Haowen Qiu)
+
+// See ../../LICENSE for clarification regarding multiple authors
+
+#include "k2/csrc/aux_labels.h"
+
+#include <numeric>
+#include <vector>
+
+#include "glog/logging.h"
+#include "k2/csrc/fsa.h"
+
+namespace k2 {
+
+void MapAuxLabels1(const AuxLabels &labels_in,
+                   const std::vector<int32_t> &arc_map, AuxLabels *labels_out) {
+  CHECK_NOTNULL(labels_out);
+  auto &start_pos = labels_out->start_pos;
+  auto &labels = labels_out->labels;
+  start_pos.clear();
+  labels.clear();
+
```

Comment:
```cpp
start_pos.reserve(arc_map.size() + 1);
labels.reserve(labels_in.labels.size());
```

Inside the `for` loop, it would be better to use

```cpp
labels.insert(labels.end(), labels_in.start_pos.begin() + arc_index,
                            labels_in.start_pos.begin() + arc_index + 1);
```

We can assume that none of the labels are epsilons.

Reply:
We can invoke `labels.resize(num_labels)` after the `for` loop in case
some of the arcs are removed from the output fsa.

Reply:
will remove `DCHECK_NE(label, kEpsilon)` and use insert, I prefer to only reserve for start_pos here (we may always allocate too much memory (even if we resize at last) for most of FSA algorithms).